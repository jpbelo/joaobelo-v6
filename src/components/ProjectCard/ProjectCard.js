import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import ToolsList from 'components/ToolsList'

const Container = styled.div`
  margin-bottom: 30px;
  h2 {
    margin-bottom: 7px;
    font-size: 15px;
    span {
      font-size: 12px;
      color: grey;
      vertical-align: top;
      margin-left: 5px;
    }
  }
  p {
    line-height: 1.6;
    font-size: 13px;
  }
  & > a {
    display: inline-block;
    margin-top: 10px;
    font-size: 13px;
    transition: color 0.2s ease-in-out;
    &[href*='http']:hover {
      color: #fc3c3c;
      opacity: 1;
    }
  }
  & > span {
    font-size: 11px;
    display: block;
    color: grey;
  }
`
const Decommissioned = styled.p`
  margin-top: 10px;
  color: #fc3c3c;
`

export default class ProjectCard extends Component {
  render() {
    const { status, name, type, description, tools, external_url } = this.props
    return (
      <Container>
        <h2>
          {name} <span>{type}</span>
        </h2>
        <p dangerouslySetInnerHTML={{ __html: description }} />
        <ToolsList toolsList={tools} />
        {status === 'online' ? (
          <a rel="noopener noreferrer" target="_blank" href={external_url}>
            â€” project link
          </a>
        ) : (
          <Decommissioned>[Project decommissioned]</Decommissioned>
        )}
      </Container>
    )
  }
}

ProjectCard.propTypes = {
  id: PropTypes.number.isRequired,
  status: PropTypes.oneOf(['online', 'decommissioned', 'hidden']),
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  tools: PropTypes.array.isRequired,
  external_url: PropTypes.string.isRequired,
}

ProjectCard.defaultProps = {
  status: 'online',
}
