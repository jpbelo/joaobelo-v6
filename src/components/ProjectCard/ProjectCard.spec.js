import React from 'react'
import 'jest-styled-components'

import { renderWithTheme } from 'helpers/renderWithTheme'
import ProjectCard from './ProjectCard'

const mockProps = {
  id: 35,
  name_id: 'joaobelo',
  name: 'joaobelo.pt (v6)',
  tags: 'JoÃ£o Belo',
  type: 'website',
  external_url: 'https://github.com/jpbelo/joaobelo-v6',
  tools: 'ReactJS, styled components',
  description: '(this actual website) project designed and developed by myself',
  date: '2019-03-01',
}

test('Project Card display correctly', () => {
  let tree = renderWithTheme(<ProjectCard {...mockProps} />).toJSON()
  expect(tree).toMatchSnapshot()
})
