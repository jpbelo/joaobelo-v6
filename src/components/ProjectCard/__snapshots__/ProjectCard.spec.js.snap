// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Card display correctly 1`] = `
.c2 {
  font-size: 11px;
  color: hsl(0,0%,50%);
}

.c2:not(:last-child)::after {
  content: ', ';
}

.c2[href^='http'] {
  color: hsl(0,0%,40%);
}

.c2[href^='http']:hover {
  color: hsl(0,97%,61%);
}

.c1 {
  margin-top: 10px;
}

.c0 {
  margin-bottom: 30px;
}

.c0 h2 {
  margin-bottom: 7px;
  font-size: 15px;
}

.c0 h2 span {
  font-size: 12px;
  color: grey;
  vertical-align: top;
  margin-left: 5px;
}

.c0 p {
  line-height: 1.6;
  font-size: 13px;
}

.c0 > a {
  display: inline-block;
  margin-top: 10px;
  font-size: 13px;
  -webkit-transition: color 0.2s ease-in-out;
  transition: color 0.2s ease-in-out;
}

.c0 > a:hover {
  color: #fc3c3c;
  opacity: 1;
}

.c0 > span {
  font-size: 11px;
  display: block;
  color: grey;
}

<div
  className="c0"
>
  <h2>
    joaobelo.pt (v6)
     
    <span>
      website
    </span>
  </h2>
  <p
    dangerouslySetInnerHTML={
      Object {
        "__html": "(this actual website) project designed and developed by myself",
      }
    }
  />
  <div
    className="c1"
  >
    <a
      className="c2"
      href="https://reactjs.org/"
      target="_blank"
    >
      ReactJS
    </a>
    <span
      className="c2"
    >
      styled components
    </span>
  </div>
  <a
    href="https://github.com/jpbelo/joaobelo-v6"
    rel="noopener noreferrer"
    target="_blank"
  >
    â€” project link
  </a>
</div>
`;
